src/controllers/nodes/nodes_controller.go

```diff
@@ -10,230 +10,192 @@ import (
 	"github.com/Dynatrace/dynatrace-operator/src/dtclient"
 	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/schema"
-	"sigs.k8s.io/controller-runtime/pkg/cache"
+	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/event"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-const (
-	cacheName = "dynatrace-node-cache"
-)
+func Add(mgr manager.Manager, _ string) error {
+	return NewReconciler(mgr).SetupWithManager(mgr)
+}
+
+func (controller *NodesController) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.Node{}).
+		WithEventFilter(nodeDeletionPredicate(controller)).
+		Complete(controller)
+}
+func nodeDeletionPredicate(controller *NodesController) predicate.Predicate {
+	return predicate.Funcs{
+		DeleteFunc: func(deleteEvent event.DeleteEvent) bool {
+			controller.reconcileNodeDeletion(deleteEvent.Object.GetName())
+			return false
+		},
+	}
+}
 
-var unschedulableTaints = []string{"ToBeDeletedByClusterAutoscaler"}
+// blank assignment to verify that NodesController implements reconcile.Reconciler
+var _ reconcile.Reconciler = &NodesController{}
```

Comment:
nitpick: not necessary here as you pass it to the manager in line 33, and if the controller didn't fulfil the `Reconciler` interface then your IDE would warn you